bench_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+)
+
+func newBenchmarkObject() interface{} {
+	data := struct {
+		Object map[string]interface{}
+		Items  []interface{}
+	}{
+		Object: map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "PodList",
+		},
+		Items: []interface{}{},
+	}
+	for i := 0; i < 1000; i++ {
+		item := struct {
+			Object map[string]interface{}
+		}{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"name":      fmt.Sprintf("pod%d", i),
+					"namespace": "ns",
+					"labels": map[string]interface{}{
+						"first-label":  "12",
+						"second-label": "label-value",
+					},
+				},
+			},
+		}
+		data.Items = append(data.Items, item)
+	}
+	return data
+}
+
+func newBenchmarkYAML() ([]byte, error) {
+	return yaml.Marshal(newBenchmarkObject())
+}
+
+func BenchmarkMarshal(b *testing.B) {
+	obj := newBenchmarkObject()
+	b.ResetTimer()
+	b.ReportAllocs()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			result, err := Marshal(obj)
+			if err != nil {
+				b.Errorf("error marshaling YAML: %v", err)
+			}
+			b.SetBytes(int64(len(result)))
+		}
+	})
+}
+
+func BenchmarkUnmarshal(b *testing.B) {
+	yamlBytes, err := newBenchmarkYAML()
+	if err != nil {
+		b.Fatalf("error initializing YAML: %v", err)
+	}
+	b.ResetTimer()
+	b.ReportAllocs()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			var result interface{}
+			if err = Unmarshal(yamlBytes, &result); err != nil {
+				b.Errorf("error unmarshaling YAML: %v", err)
+			}
+		}
+	})
+	b.SetBytes(int64(len(yamlBytes)))
```

Comment:
should this be inside the pb.Next() loop?

Reply:
It can be, but doesn't really matter.  My understanding is `b.SetBytes` [is not cumulative](https://cs.opensource.google/go/go/+/master:src/testing/benchmark.go;l=171?q=setbytes&sq=&ss=go%2Fgo), so it would just be overwriting it inside the loop.

In the other tests, I called SetBytes inside the loop because that is where the number of bytes was available (only after marshaling occurred).  If anything I should probably take it out of the loop in the other tests, to avoid confusion.

Reply:
Moved other calls outside of loop for consistency